### PR TITLE
homework5

### PR DESCRIPTION
**Pull request by najahizquierdo**
....very very bad but it works when ran from the terminal!  had to get rid of alot of features and rankings because it isnt working well (YET) :(

### CODE REVIEW

**Comment by najahizquierdo** [comment]
@brian004900 @dafrenchyman

File: pytosql.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                , bc.Hit/bc.atBat as batting_avg
+                , bc.Groundout/bc.Flyout as ground_fly_ratio
+                , (bc.Hit+bc.Walk+bc.Hit_By_Pitch)/(bc.atBat+bc.Walk+bc.Hit_By_Pitch+bc.Sac_Fly) as on_base_percentage
+            FROM game g
+            JOIN inning i ON g.game_id = i.game_id
+            JOIN team_batting_counts bc ON g.game_id = bc.game_id
+            JOIN boxscore b ON g.game_id = b.game_id
+            JOIN team_pitching_counts t ON g.game_id = t.game_id
+            GROUP BY t.game_id
+    """
```
Comment (brian004900) [inline pytosql.py:24]: I am not able to run this SQL script.
"[2022-11-17 22:07:28] [22007][1292] (conn=177) Truncated incorrect DOUBLE value: 'H'" showed

File: pytosql.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                , (bc.Strikeout/bc.toBase) as strikeout_to_walk_ratio
+                , b.winddir as wind_direction
+                , bc.Hit/bc.atBat as batting_avg
+                , bc.Groundout/bc.Flyout as ground_fly_ratio
+                , (bc.Hit+bc.Walk+bc.Hit_By_Pitch)/(bc.atBat+bc.Walk+bc.Hit_By_Pitch+bc.Sac_Fly) as on_base_percentage
+            FROM game g
+            JOIN inning i ON g.game_id = i.game_id
+            JOIN team_batting_counts bc ON g.game_id = bc.game_id
+            JOIN boxscore b ON g.game_id = b.game_id
+            JOIN team_pitching_counts t ON g.game_id = t.game_id
```
Comment (brian004900) [inline pytosql.py]: good to create features on both batting and pitching table

File: midterm.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from plotly import express as px
+from plotly import graph_objects as go
+from sklearn.ensemble import RandomForestClassifier
+
+
+def correlation_metrics(x):
+    pearsoncorr = x.corr(method="pearson")
+    fig = px.imshow(pearsoncorr)
+    fig.show()
+
```
Comment (brian004900) [inline midterm.py:17]: Able to generate correlation_metrics plot.

File: midterm.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    for a in numerical_cols:
+        feature_data = df[a]
+        predictor = statsmodels.api.add_constant(feature_data)
+        linear_regression_model = statsmodels.api.OLS(response, predictor)
+        linear_regression_model_fitted = linear_regression_model.fit()
+        print(linear_regression_model_fitted.summary())
+        t_value = round(linear_regression_model_fitted.tvalues[1], 6)
+        p_value = "{:.6e}".format(linear_regression_model_fitted.pvalues[1])
+        fig1 = px.histogram(
```
Comment (brian004900) [inline midterm.py]: Every fig(fig1, fig_2) successfully showed on the web page.

File: midterm.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    cont_resp = pd.DataFrame(CONT_COLUMNS)
+    print(cont_resp)
+    cat_resp = pd.DataFrame(CAT_COLUMNS)
+    print(cat_resp)
+    cont_resp_T = cont_resp.T
+    correlation_metrics(cont_resp_T)
+
+    for a in numerical_cols:
+        feature_data = df[a]
+        predictor = statsmodels.api.add_constant(feature_data)
```
Comment (brian004900) [inline midterm.py:116]: response here is category.

File: assignment5baseball.sql [hunk 6]
```diff
@@ -0,0 +1,52 @@
+#this isnt implemented into the python code yet so you dont have to run this right now (only for file that i turn into the professor)
```
Comment (brian004900) [inline assignment5baseball.sql]: ignored this file in pr review based on this note.

File: midterm.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            print(f"{i} is continuous")
+            p = 1
+            cont_predictors.append(i)
+    CONT_COLUMNS = list()
+    CAT_COLUMNS = list()
+    for colName, i in df.iteritems():
+        if colName in cat_predictors:
+            CAT_COLUMNS.append(i)
+        else:
+            CONT_COLUMNS.append(i)
```
Comment (brian004900) [inline midterm.py]: good to determine predictor types and list them to check every feature

File: pytosql.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    #     , a.away_strikeout_to_walk_ratio
+    #     , a.away_batting_avg
+    #     , a.away_on_base_percentage
+    # FROM game_stats g
+    # JOIN home_stats h ON g.game_id = h.game_id
+    # JOIN away_stats a ON g.game_id = a.game_id
+    # GROUP BY g.game_id
+    baseball_df = pandas.read_sql_query(pitching, sql_engine)
+    response = baseball_df["binary_winner"]
+    df_processing(baseball_df, response)
```
Comment (brian004900) [inline pytosql.py]: It is good here to process a function in a different py file, this makes the script more concise.

File: assignment5baseball.sql [hunk 9]
```diff
@@ -1,52 +1,93 @@
-#this isnt implemented into the python code yet so you dont have to run this right now (only for file that i turn into the professor)
-CREATE OR REPLACE TABLE game_stats (
+USE BASEBALL;
```
Comment (najahizquierdo) [inline assignment5baseball.sql]: Completely changed my SQL :|

File: midterm.py [hunk 10]
```diff
@@ -1,109 +1,192 @@
...
+        fig_2.show()
 
+def linear_regression(df, numerical_cols, response):
     for a in numerical_cols:
         feature_data = df[a]
         predictor = statsmodels.api.add_constant(feature_data)
-        linear_regression_model = statsmodels.api.OLS(response, predictor)
-        linear_regression_model_fitted = linear_regression_model.fit()
-        print(linear_regression_model_fitted.summary())
+        linear_regression_model_fitted = statsmodels.api.OLS(response, predictor).fit()
```
Comment (najahizquierdo) [inline midterm.py:117]: I couldnt get the Logit to work :\

**Pull request merged**